src/source/reporting.rst

```diff
@@ -0,0 +1,285 @@
+
+Filters
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. _filters:
+
+To request a subset of your data, use filters. For example, you can filter on particular category, region, or ad ID. Filters can currently be used on any of the supported dimensions (except for ``am:date``). In the request query, they should be provided as an array of filter clauses which filters out the requested data on particular field values. Each filter clause is of the following form:
+
+.. code:: javascript
+
+        {
+                "field":   <string>
+                "operator":  <string>
+                "value":  [<long>, ...]
+        }
+
+Each filter in the list adds more restriction on the result, i.e., there is implicit *AND* operation between them.
+The only valid ``operator`` at the moment is the set operator ``in``, and requires an array of values to be passed in the ``value`` field of the filter. For example:
+
+.. code:: javascript
+
+    "filters": [
+        {
+            "field":"am:adID",
+            "operator":"in",
+            "value": [1111,2222,3333]
+        },
+        {
+            "field":"am:regionID",
+            "operator":"in",
+            "value": [1234, 5678]
+        }
+    ]
+
+We plan to extend support for filters on metrics, as well as more comparison operators, if there is demand for it.
+
+.. _Sorts:
+
+Sorting
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+To sort the resulting data, use the **sorts** top-level query field. Sorts is an array of sort clauses, each of the following form:
+
+.. code:: javascript
+
+    {
+        "field":        <string>
+        "direction":    <string>
+    }
+
+
+A field can be either a valid metric or a valid dimension. It is only permitted to sort on the fields that are requested obviously. There are two directions for sorting: ``asc`` (ascending) and ``desc`` (descending). 
+The following example sorts first on date in descending order, followed by the ad ID in ascending order.
+
+.. code:: javascript
+
+    {
+        "sorts": [
+            {
+                "field":"am:date",
+                "direction":"desc"
+     
+            },
+            {
+                "field":"am:adID",
+                "direction":"asc"
+            }
+        ]
+    }
+
+Pagination with offset and limit
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Depending on the amount and shape of the requested data, it is worthwhile to paginate the results, for efficiency. Use the top-level ``offset`` and ``limit`` fields of the request query, with their standard meanings.
+
+
+Enrichment
+~~~~~~~~~~~~~~~~~~~~~~~~~
+It is possible to enrich the reports with "handy" ad-related data for which only the present-moment values are available, such as the ad title, or category description. Requesting these fields makes sense **only** if ``am:adID`` is in the requested dimensions. The supported fields are listed below:
```

Comment:
should we build validation on requests having these fields but no 'am:adID' in the dimensions?

Reply:
we should... I will do today